### DISCUSSION TITLE
Camera status synchronization problem in VtkRemoteLocalView rendering mode switching

### DISCUSSION THREAD

**Discussion by Yukeaaa** [Kitware/trame#371] [Q&A] [answered]
Dear Community,

I'm currently learning VTK and Trame and I'm running into issues related to camera state synchronization when using `VtkRemoteLocalView` for rendering.

The program provides users with a button to switch rendering modes on the front end. The following problems were discovered during use.
When mode is `local`, the server will synchronize the camera status during user interaction. When the mode is switched from `local` to `remote`, the synchronization status can be correct.
But when the mode is `remote`, the user switches to `local` after some interactions, and the camera parameters are still in their previous state. (i.e. the camera status when the last switch was from `remote` to `local`)

Could you tell me how to synchronize camera status? Thank you very much for your help!

```
with SinglePageLayout(server) as layout:
    layout.icon.click = ctrl.view_reset_camera
    layout.title.set_text("hybrid render")

    with layout.toolbar:
        html.Div("{{ mode }}")
        vuetify.VSpacer()
        vuetify.VCheckbox(
            v_model=("mode", mode),
            off_icon="mdi-image",
            on_icon="mdi-rotate-3d",
            true_value="local",
            false_value="remote",
            dense=True,
            hide_details=True,
        )
        vuetify.VSlider(
            v_model=("quality", update_reset_quality),
            min=0,
            max=100,
            step=10,
            hide_details=True,
            dense=True,
            style="max-width: 300px",
        )
        vuetify.VDivider(vertical=True, classes="mx-2")
        with vuetify.VBtn(icon=True, click=update_reset_quality()):
            vuetify.VIcon("mdi-undo-variant")

    with layout.content:
        with vuetify.VContainer(
            fluid=True,
            classes="pa-0 fill-height",
        ):
            # view = vtk.VtkRemoteView(renWin)
            # view = vtk.VtkLocalView(renWin)
            view = vtk.VtkRemoteLocalView(view=renWin,
                                          namespace="hybrid",
                                          mode=("mode",),
                                          disable_auto_switch="False",
                                          interactive_ratio=1,
                                          interactive_quality=("quality",))
            # Force image to be pushed to client
            ctrl.view_update = view.update
            ctrl.view_reset_camera = view.reset_camera
```

**Comment by jourdain** [comment, marked as answer]
You can either call that [method](https://github.com/Kitware/trame-vtk/blob/master/trame_vtk/widgets/vtk/common.py#L475) once for auto-synch or call [that method](https://github.com/Kitware/trame-vtk/blob/master/trame_vtk/widgets/vtk/common.py#L567) yourself when you switch mode.

**Reply by jourdain** [reply]
You can find an [example here](https://github.com/Kitware/trame-vtk/blob/master/examples/vue3/cone-toggle.py#L102)

**Reply by Yukeaaa** [reply]
It worked! Thank you very much!
